src/axom/core/Array.hpp

```diff
@@ -649,22 +700,34 @@ class Array : public ArrayBase<T, DIM, Array<T, DIM, SPACE>>
    * \post size() == num_elements
    * \post getResizeRatio() == DEFAULT_RESIZE_RATIO
    */
-  void initialize(IndexType num_elements, IndexType capacity);
+  void initialize(IndexType num_elements,
+                  IndexType capacity,
+                  bool default_construct = true);
```

Comment:
Suggestion (minor): it might be clearer if the new parameter were named `should_default_construct`